ignite/handlers/checkpoint.py

```diff
@@ -353,6 +362,33 @@ def _setup_checkpoint(self) -> dict:
             checkpoint[k] = obj.state_dict()
         return checkpoint
 
+    @staticmethod
+    def setup_filename_pattern(
+        filename_prefix, with_score, with_score_name, with_global_step_transform, filename_pattern=None,
+    ):
```

Comment:
We need to provide a docstring for this public method, remove `filename_pattern=None` from args and update the code accordingly. 
We also need annotate input args (`typing`)

Reply:
```
    @staticmethod
    def setup_filename_pattern(
        filename_prefix: str = "",
        with_score: bool = False,
        with_score_name: bool = False,
        with_global_step_transform: bool = False,
        filename_pattern: str = False,
    ) -> str:
```
Is this ok @vfdev-5

Reply:
The typehint for `filename_pattern` is wrong 
```
filename_pattern: Optional[str] = None
```

Reply:
@sdesrozis we do not need this arg at all. See my message above, please

Reply:
Sorry, @vfdev-5 I was a bit confused, 
> Checkpoint.setup_filename_pattern(
>                 filename_prefix=self.filename_prefix,
>                 with_score=self.score_function is None,
>                 with_score_name=self.score_name is None,
>                 with_global_step_transform=self.global_step_transform is None,
>             )

Are you suggesting that we don't require the filename_pattern to be passed but rather checked before calling `setup_filename_pattern`.
**def __call__**
```
if self.filename_pattern is None:
    filename_pattern = self.setup_filename_pattern(
        filename_prefix=self.filename_prefix,
        with_score=self.score_function is None,
        with_score_name=self.score_name is None,
        with_global_step_transform=global_step is None,
    )
else:
    filename_pattern = self.filename_pattern
```
**def setup_filename_pattern**
```
def setup_filename_pattern(
    filename_prefix: str = "",
    with_score: bool = False,
    with_score_name: bool = False,
    with_global_step_transform: bool = False,
) -> Union[str, None]:
```
is this what you are suggesting?

Reply:
Yes, something like that. Note that `setup_filename_pattern` returns `str` and not `Union`.
```python
def setup_filename_pattern(...) -> str
```

Reply:
you are right @vfdev-5 will update this as well.